src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java

```diff
@@ -1,16 +1,28 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import java.util.NoSuchElementException;
+import mate.academy.hibernate.relations.dao.CountryDao;
+import mate.academy.hibernate.relations.dao.impl.CountryDaoImpl;
 import mate.academy.hibernate.relations.model.Country;
 import mate.academy.hibernate.relations.service.CountryService;
+import org.hibernate.SessionFactory;
 
 public class CountryServiceImpl implements CountryService {
+    private CountryDao countryDao;
+
+    public CountryServiceImpl(SessionFactory sessionFactory) {
```

Comment:
The constructor should accept `CountryDao` as a parameter instead of `SessionFactory`. This would allow for better separation of concerns and make the service layer not dependent on a particular implementation of the DAO layer.